Other/SelectableCollection.cs

```diff
@@ -0,0 +1,194 @@
+
+        public SelectableCollection(List<T> list) : base(list) { }
+
+        #endregion
+
+        #region Override List
+
+        public new void Add(T Item)
+        {
+            base.Add(Item);
+            Select(OnAdd);
+        }
+
+        /// <summary>
+        /// Removes the first occurrence of a specific object from the Collections.
+        /// </summary>
+        /// <param name="Item">Item which should be deleted</param>
+        /// <returns>true if successfully</returns>
+        public new bool Remove(T Item) => Remove(Item, OnRemoving);
+
+        /// <summary>
+        /// Removes the first occurrence of a specific object from the Collections.
+        /// </summary>
+        /// <param name="Item">Item which should be deleted</param>
+        /// <param name="selectProperties">is used instead of OnRemoving</param>
+        /// <returns>true if successfully</returns>
+        public bool Remove(T Item, SelectPropertie selectProperties)
+        {
+            if (!IsSelected(Item))
+                return base.Remove(Item);
+
+            int i = IndexOf(Selected);
+            base.Remove(Item);
+            Select(i, selectProperties);
+
+            return true;
+        }
+
+        public new bool RemoveAt(int index) => RemoveAt(index, OnRemoving);
+        public bool RemoveAt(int index, SelectPropertie selectProperties) => Remove(Items[index], selectProperties);
+
+        public bool RemoveSelected() => RemoveSelected(OnRemoving);
+        public bool RemoveSelected(SelectPropertie selectProperties)
+        {
+            if (!ValidSelection)
+                return false;
+
+            return Remove(Selected, selectProperties);
+        }
+
+        public new void Clear()
+        {
+            base.Clear();
+            Selected = default;
+        }
+
+        #endregion
+
+        #region Select
+
+        public bool Select(T Item)
+        {
+            Selected = Item;
+            return IsSelected(Item);
+        }
+
+        /// <summary>
+        /// selects the item that is at a relative position
+        /// </summary>
+        /// <param name="selectPropertie">specify the position</param>
+        /// <returns>true if successfully</returns>
+        public bool Select(SelectPropertie selectPropertie) => Select(IndexOf(Selected), selectPropertie);
+        private bool Select(int i, SelectPropertie selectPropertie)
+        {
+            if (Count == 0)
+            {
+                Selected = default;
+                return false;
+            }
+
+            switch (selectPropertie)
+            {
+                case SelectPropertie.First:
+                    Selected = Items[0];
+                    break;
+                case SelectPropertie.Last:
+                    Selected = Items[Count - 1];
+                    break;
+                case SelectPropertie.Previous:
+                    if (i <= 0) goto case SelectPropertie.Last;
```

Comment:
As mentioned before, not a fan of `goto`s.  This usage isn't horrible but I don't like the idea of putting it in the code base to promote future usage.

Reply:
`goto `is totally ok with switch in my opinion.
` if (i == -1) Select(SelectPropertie.First);`
I'll see how I can change it.